lib/models/device.js

```diff
@@ -2711,6 +2711,79 @@ const getDeviceModel = function (deps, opts) {
 			});
 		},
 
+		/**
+		 * @summary Set a specific device to run a particular supervisor release
+		 * @name setSupervisorRelease
+		 * @public
+		 * @function
+		 * @memberof balena.models.device
+		 *
+		 * @description Configures the device to run a particular supervisor release.
+		 *
+		 * @param {String|Number} uuidOrId - device uuid (string) or id (number)
+		 * @param {String|Number} supervisorVersionOrId - the version of a released supervisor (string) or id (number)
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.device.setSupervisorRelease('7cf02a6', 'v10.8.0').then(function() {
+		 * 	...
+		 * });
+		 *
+		 * @example
+		 * balena.models.device.setSupervisorRelease(123, 'v11.4.14').then(function() {
+		 * 	...
+		 * });
+		 *
+		 * @example
+		 * balena.models.device.setSupervisorRelease('7cf02a6', 123, function(error) {
+		 * 	if (error) throw error;
+		 * 	...
+		 * });
+		 */
+		setSupervisorRelease: async (uuidOrId, supervisorVersionOrId) => {
+			let deviceId;
+			let releaseId;
+			if (isId(uuidOrId) && isId(supervisorVersionOrId)) {
+				deviceId = uuidOrId;
+				releaseId = supervisorVersionOrId;
+			} else {
+				const releaseFilterProperty = isId(supervisorVersionOrId)
+					? 'id'
+					: 'supervisor_version';
+				const device = await exports.get(uuidOrId, {
+					$expand: { is_of__device_type: { $select: 'slug' } },
+				});
+				const supervisorRelease = await pine.get({
```

Comment:
NIT: How about doing:
```suggestion
				const [supervisorRelease] = await pine.get({
```
so that we can just check for `== null` instead of `.length` ?

Reply:
Yep, that sounds better. Updated.